### PR TITLE
Fix no browser environment support

### PR DESCRIPTION
**Pull request by orzechdev**

### CODE REVIEW

File: src/api/APIProxy.ts [hunk 1]
```diff
@@ -177,10 +177,14 @@ class APIProxy {
       callback(this.isLoggedIn());
     };
 
-    window.addEventListener("auth", eventHandler);
+    if (typeof window !== "undefined") {
+      window.addEventListener("auth", eventHandler);
```
Comment (patrys) [inline src/api/APIProxy.ts:181]: Can't we use the `?.` operator here?

**Review by dominik-zeglen — CHANGES_REQUESTED** [review]
Instead of doing the same check over and over again, I believe we should export some variable that states environment based on `window === undefined`

**Review by dominik-zeglen — APPROVED** [review]
👌

File: src/consts.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export const WINDOW_EXISTS = typeof window !== "undefined";
```
Comment (patrys) [inline src/consts.ts:1]: What if we made the check the opposite and named it something like `SERVER_SIDE_RENDERING`?

File: src/consts.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+
+export const LOCAL_STORAGE_EXISTS = WINDOW_EXISTS && !!window.localStorage;
+export const CREDENTIAL_API_EXISTS =
+  WINDOW_EXISTS && !!window.PasswordCredential;
```
Comment (patrys) [inline src/consts.ts:5]: These seem too specific for generic constants.

**Review by patrys — CHANGES_REQUESTED** [review]

File: src/consts.ts (hunk 2 above)
Comment (orzechdev) [reply src/consts.ts:1]: But how do you know that when window is undefined it is rendered on server side? We may also use similar environment to React Native, which doesn't run on server.

**Review by AlicjaSzu — APPROVED** [review]

File: src/consts.ts (hunk 2 above)
Comment (patrys) [reply src/consts.ts:1]: I think for RN you need way more logic (like AsyncStorage). For that to work, we would have to split the SDK and initialize it like you initialize Apollo:

```typescript
import { createStorage } from "@saleor/browser";
import { SaleorManager } from "@saleor/sdk";

const storage = createStorage();
const client = new SaleorManager({ storage });
```

(Ignore the names, it's all just for illustrative purposes.)

**Review by patrys — APPROVED** [review]

**Pull request merged**